io/iouring-wrapper.cpp

```diff
@@ -501,21 +511,9 @@ class iouringEngine : public MasterEventEngine, public CascadingEventEngine {
         }
     }
 
-    void run_cancel_poller() {
-        while (m_cancel_poller_running) {
-            uint32_t poll_mask = evmap.translate_bitwisely(EVENT_READ);
-            int ret = async_io(&io_uring_prep_poll_add, -1, 0, m_cancel_fd, poll_mask);
-            if (ret < 0) {
-                if (errno == EINTR) {
-                    break;
-                }
-                LOG_ERROR("iouring: poll eventfd failed, `", ERRNO());
-            }
-            eventfd_t val;
-            if (eventfd_read(m_cancel_fd, &val) != 0) {
-                LOG_ERROR("iouring: read eventfd failed, `", ERRNO());
-            }
-        }
+    void run_wait_canceller() {
+        uint32_t poll_mask = evmap.translate_bitwisely(EVENT_READ);
+        async_io(&io_uring_prep_poll_multishot, -1, RingFlagsWaitCanceller, m_wait_canceller_fd, poll_mask);
     }
```

Comment:
`wait_canceller` will fall into sleep in `async_io`. Because the multishot poll needs to keep the `ioCtx` on stack for the whole time.

Reply:
actually we don't need to use async_io() to register the multishot poll, and we can set the user data as nullptr.

When we get a event with nullptr data, we know it's cancel wait.